packages/react/src/providers/treasure.tsx

```diff
@@ -66,6 +66,8 @@ export const useTreasure = () => {
 
 type Props = PropsWithChildren<Config>;
 
+let hasSetUrlParams = false;
```

Comment:
This could be done in a hook with `useRef` instead to keep it React-y

Reply:
Done